src/Cli/dotnet/commands/dotnet-tool/install/ToolInstallLocalCommand.cs

```diff
@@ -46,16 +56,25 @@ internal class ToolInstallLocalCommand : CommandBase
                                   new ToolManifestFinder(new DirectoryPath(Directory.GetCurrentDirectory()));
             _toolManifestEditor = toolManifestEditor ?? new ToolManifestEditor();
             _localToolsResolverCache = localToolsResolverCache ?? new LocalToolsResolverCache();
-            _toolLocalPackageInstaller = new ToolInstallLocalInstaller(parseResult, toolPackageDownloader);
+            _toolLocalPackageInstaller = new ToolInstallLocalInstaller(parseResult, _packageId, toolPackageDownloader);
+            _toolPackageDownloader = toolPackageDownloader;
             _allowRollForward = parseResult.GetValue(ToolInstallCommandParser.RollForwardOption);
-
-            _packageId = new PackageId(parseResult.GetValue(ToolUpdateCommandParser.PackageIdArgument));
             _allowPackageDowngrade = parseResult.GetValue(ToolInstallCommandParser.AllowPackageDowngradeOption);
-
         }
-        
 
         public override int Execute()
+        {
+            if (_all)
+            {
+                return ExecuteInstallAllCommand();
+            }
+            else
+            {
+                return ExecuteInstallCommand();
+            }
+        }
+
+        private int ExecuteInstallCommand()
```

Comment:
Similiar, the `packageId` might also need to passed in to `InstallNewTool` and `InstallToolUpdate`, as well as the `install` method in `toolLocalPackageInstaller`.

Reply:
Done! Thanks for the note! This would be super clear, concise, elegant, and fast.